src/test/fixtures/profiles/processed-profile.js

```diff
@@ -190,13 +190,17 @@ export function getProfileFromTextSamples(
   profile.threads = allTextSamples.map(textSamples => {
     // Process the text.
     const textOnlyStacks = _parseTextSamples(textSamples);
-    const funcNames = textOnlyStacks
-      // Flatten the arrays.
-      .reduce((memo, row) => [...memo, ...row], [])
-      // remove modifiers
-      .map(func => func.replace(/\[.*/, ''))
-      // Make the list unique.
-      .filter((item, index, array) => array.indexOf(item) === index);
+
+    // Flatten the textOnlyStacks into into a list of function names.
+    const funcNamesSet = new Set();
+    const removeModifiers = /\[.*/;
+    for (let i = 0; i < textOnlyStacks.length; i++) {
+      const textOnlyStack = textOnlyStacks[i];
+      for (let j = 0; j < textOnlyStack.length; j++) {
```

Comment:
I was gonna suggest `flat` with:
```js
for (const stack of textOnlyStacks.flat()) {
  funcNamesSet.add(stack.replace(removeModifiers, ''));
}
```
but it's only supported in node v11, and I don't think it's worth forcing everybody to upgrade now. Maybe in a few months...